src/network/networkd-dhcp-server.c

```diff
@@ -744,3 +749,45 @@ int config_parse_dhcp_server_address(
         network->dhcp_server_address_prefixlen = prefixlen;
         return 0;
 }
+
+int config_parse_dhcp_server_boot_server(
+                const char *unit,
+                const char *filename,
+                unsigned line,
+                const char *section,
+                unsigned section_line,
+                const char *lvalue,
+                int ltype,
+                const char *rvalue,
+                void *data,
+                void *userdata) {
+
+        Network *network = ASSERT_PTR(userdata);
+        union in_addr_union a;
+        int r;
+
+        assert(filename);
+        assert(lvalue);
+        assert(rvalue);
+
+        if (isempty(rvalue)) {
+                network->dhcp_server_boot_server = (struct in_addr) {};
+                network->dhcp_server_boot_server_name = mfree(network->dhcp_server_boot_server_name);
+                return 0;
+        }
+
+        r = in_addr_from_string(AF_INET, rvalue, &a);
+        if (r >= 0) {
+                network->dhcp_server_boot_server = a.in;
+                return 0;
+        }
+
+        r = dns_name_is_valid(rvalue);
+        if (r <= 0) {
+                log_syntax(unit, LOG_WARNING, filename, line, r,
+                           "Specified invalid boot server address or name, ignoring assignment: %s", rvalue);
+                return 0;
+        }
+
+        return free_and_strdup_warn(&network->dhcp_server_boot_server_name, rvalue);
```

Comment:
conversely, should also do `network->dhcp_server_boot_server = (struct in_addr) {};` here